Core/RoverControllerPresets/DriveControllers/DirectDriveController.cs

```diff
@@ -14,21 +16,62 @@ public RoverControl CalculateMoveVector()
 			Convert.ToSingle(LocalSettings.Singleton.Joystick.Deadzone)
 		);
 
-		Vector2 vec = new(
-			Input.GetAxis("rover_move_backward", "rover_move_forward"),
-			Input.GetAxis("rover_move_right", "rover_move_left")
-		);
+		Vector3 vec;
+
+		Mode = KinematicMode.Ackermann;
+		Mode = Input.IsActionPressed("crab_mode") ? KinematicMode.Crab : Mode;
+		Mode = Input.IsActionPressed("spinner_mode") ? KinematicMode.Spinner : Mode;
+		Mode = Input.IsActionPressed("ebrake_mode") ? KinematicMode.EBrake : Mode;
+
+		switch (Mode)
+		{
+			case KinematicMode.Ackermann:
+				vec = new(
+					Input.GetAxis("rover_move_backward", "rover_move_forward"),
+					Input.GetAxis("rover_move_right", "rover_move_left"),
+					0);
+				break;
+			case KinematicMode.Crab:
+				vec = new(
+					Input.GetAxis("rover_move_backward", "rover_move_forward"),
+					Input.GetAxis("rover_move_right", "rover_move_left"),
+					Input.GetAxis("rover_move_down", "rover_move_up")
+				);
+				break;
+			case KinematicMode.Spinner:
+				vec = new(
+					Input.GetAxis("rover_move_backward", "rover_move_forward"),
+					0f,
+					0f
+				);
+				break;
+			case KinematicMode.EBrake:
+				vec = new(0, 0, 0); //todo wheels in X pattern?
+				break;
+			default:
+				vec = new(0, 0, 0);
+				break;
+		}
 
-		if (LocalSettings.Singleton.SpeedLimiter.Enabled)
+
+		if (LocalSettings.Singleton.SpeedLimiter.Enabled) 
 			vec.X *= LocalSettings.Singleton.SpeedLimiter.MaxSpeed;
+		
+		
 
 		vec.X = Mathf.IsEqualApprox(vec.X, 0f, joyDeadZone) ? 0 : vec.X;
 		vec.Y = Mathf.IsEqualApprox(vec.Y, 0f, joyDeadZone) ? 0 : vec.Y;
+		vec.Z = Mathf.IsEqualApprox(vec.Z, 0f, joyDeadZone) ? 0 : vec.Z;
 
 		if (Input.IsActionPressed("camera_zoom_mod"))
 			vec.X /= 4f;
 
-		return RoverControlVec2Extension.FromVector2(vec);
+		return RoverControlVec3Extension.FromVector3(vec);
```

Comment:
What's that form... We use extensions to have thaat:
```suggestion
		return vector.ToRoverControl();
```
_ToRoverControl - method name addressed in other suggestion block_

Reply:
same as above